### ISSUE TITLE
Not all SEARCH/REPLACE pattern are applied; commit only contains the first one

### ISSUE THREAD

**Issue by sotte** [paul-gauthier/aider#1260]
### Issue

Not all search/replace pattern are applied, in most(?) cases only the first search/replace pattern is applied.

Here is an example of a good refactoring of aider:
![image](https://github.com/user-attachments/assets/57c5b76e-dc2f-4c6f-9e38-f967ff56aaca)

The resulting commit only contains the first diff (`git show`):
![image](https://github.com/user-attachments/assets/995f060f-e420-4cd6-bfd7-d2c1342a4161)

Just a hunch: maybe the problem is that I use [delta](https://github.com/dandavison/delta)? I'll follow up on this.

### Version and model info

```
Aider v0.51.0
Main model: gpt-4o with diff edit format
Weak model: gpt-4o-mini
```

**Comment by ozapinq** [comment]
Hi! Could you provide an example file and prompt used? Which model do you use? Did you try the latest version of aider?

**Comment by sotte** [comment]
I updated the model version in the original post.

I can't remember the exact prompt, but I repeated it with a prompt that is very close. The proposed changes look identical, but now ALL search/replace patterns contain the `<<<<<<< SEARCH` prefix, not just the first one. 
![image](https://github.com/user-attachments/assets/3f37b884-cc99-4558-acf3-17e9bdd984a7)

Maybe the model just got confused?

I'll update aider later and see if this problem keeps popping up.
